tests/test_nodes.py

```diff
@@ -21,6 +21,7 @@
 )
 from wsimod.nodes.storage import Storage
 from wsimod.nodes.waste import Waste
+import os
```

Comment:
```suggestion
from pathlib import Path
```

Reply:
Tried but still not passed ...

Reply:
It looks like that's just the pre commit since my code suggestion doesn't fit the formatting rules. If you pull, run `pre-commit run --all-files`, add, commit, push, then hopefully it will pass

Reply:
passed for the qa but not for windows-latest 3.10, here is the error message:
'Run python -m pytest
C:\hostedtoolcache\windows\Python\3.[10](https://github.com/ImperialCollegeLondon/wsi/actions/runs/10190065866/job/28189182286#step:5:11).11\x64\python.exe: No module named pytest
Error: Process completed with exit code 1.'

Reply:
what in the world... super weird... I'm happy to approve - not sure if @dalonsoa may have views though - will see when he's back

Reply:
For Python 3.9 and 3.11, the dependencies are all installed fine and you get at the end of the `Install and build` step in the CI file something like:
```
Installing collected packages: pytz, distlib, wheel, tzdata, six, ruff, pyyaml, pyproject-hooks, pyparsing, pluggy, platformdirs, pillow, pathspec, packaging, numpy, nodeenv, mypy-extensions, kiwisolver, iniconfig, identify, fonttools, filelock, dill, cycler, coverage, colorama, cfgv, certifi, attrs, virtualenv, tqdm, shapely, python-dateutil, pytest, pyproj, contourpy, click, build, pytest-cov, pre-commit, pip-tools, pandas, matplotlib, cligj, click-plugins, black, fiona, geopandas
Successfully installed attrs-23.2.0 black-23.10.1 build-1.0.3 certifi-2023.11.17 cfgv-3.4.0 click-8.1.7 click-plugins-1.1.1 cligj-0.7.2 colorama-0.4.6 contourpy-1.2.0 coverage-7.3.2 cycler-0.12.1 dill-0.3.7 distlib-0.3.7 filelock-3.13.1 fiona-1.9.5 fonttools-4.47.0 geopandas-0.14.2 identify-2.5.31 iniconfig-2.0.0 kiwisolver-1.4.5 matplotlib-3.8.2 mypy-extensions-1.0.0 nodeenv-1.8.0 numpy-1.26.2 packaging-23.2 pandas-2.1.3 pathspec-0.11.2 pillow-10.2.0 pip-tools-7.3.0 platformdirs-3.11.0 pluggy-1.3.0 pre-commit-3.5.0 pyparsing-3.1.1 pyproj-3.6.1 pyproject-hooks-1.0.0 pytest-7.4.3 pytest-cov-4.1.0 python-dateutil-2.8.2 pytz-2023.3.post1 pyyaml-6.0.1 ruff-0.1.4 shapely-2.0.2 six-1.16.0 tqdm-4.66.1 tzdata-2023.3 virtualenv-20.24.6 wheel-0.41.3
```

But for Python 3.10 with Windows in the `pull_request` scenario you get:

```
...
Downloading colorama-0.4.6-py2.py3-none-any.whl (25 kB)
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    unknown package:
        Expected sha256 5b4c6a3df53bee8f85bb46685562b21b43346be1fe96419f18f70fa1ab8c561c
             Got        07ce221ab790f1ee6963ea46b7aef0e53e18dbc9c77d48075e7e4363dd02213c
...
```

And then, a bit further down, 

```
Installing collected packages: pytz, tzdata, six, PyYAML, numpy, dill, colorama, tqdm, python-dateutil, pandas, wsimod
Successfully installed PyYAML-6.0.1 colorama-0.4.6 dill-0.3.8 numpy-2.0.1 pandas-2.2.2 python-dateutil-2.9.0.post0 pytz-2024.1 six-1.16.0 tqdm-4.66.4 tzdata-2024.1 wsimod-0.4.0
```

So, indeed, `pytest` is not installed - and neither many other packages. It has nothing to do with using `pathlib` or with `pre-commit` but with that hash that is not what it should. 

I think it must have been a glitch or something because I've re-run the failed job and now it passes fine.

When this kind of very weird things happen, just re-run the jobs a couple of times - if some time in between - because GitHub actions, like any other software, do have errors and temporary issues that fix themselves if given some time.